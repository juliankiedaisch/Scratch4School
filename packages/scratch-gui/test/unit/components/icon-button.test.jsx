import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom';
import IconButton from '../../../src/components/icon-button/icon-button';

describe('IconButtonComponent', () => {
    const defaultProps = {
        img: 'imgSrc',
        title: <div>Text</div>,
        onClick: () => {},
        className: 'custom-class-name'
    };

    test('renders with all props correctly', () => {
        const {container} = render(<IconButton {...defaultProps} />);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('triggers callback when clicked', () => {
        const onClick = jest.fn();

        const {container} = render(
            <IconButton
                {...defaultProps}
                onClick={onClick}
            />
        );

        const button = container.querySelector('div[role="button"]');

        fireEvent.click(button);

        expect(onClick).toHaveBeenCalledTimes(1);
    });

    test('does not trigger callback when not clicked', () => {
        const onClick = jest.fn();

        render(<IconButton
            {...defaultProps}
            onClick={onClick}
        />);
        
        expect(onClick).toHaveBeenCalledTimes(0);
    });
});
