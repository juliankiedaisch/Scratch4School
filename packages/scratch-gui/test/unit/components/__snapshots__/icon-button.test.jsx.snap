// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonComponent renders with all props correctly 1`] = `
<div
  class="custom-class-name"
  role="button"
>
  <img
    draggable="false"
    src="imgSrc"
  />
  <div>
    <div>
      Text
    </div>
  </div>
</div>
`;
